type_lens/type_view.py

```diff
@@ -219,14 +219,14 @@ def is_subtype_of(self, typ: Any | tuple[Any, ...], /) -> bool:
             Whether the annotation is a subtype of the given type(s).
         """
         if self.origin:
-            if self.origin in UNION_TYPES:
+            if self.is_union:
                 return all(t.is_subtype_of(typ) for t in self.inner_types)
 
-            return self.origin not in UNION_TYPES and issubclass(self.origin, typ)
+            return issubclass(self.origin, typ)
 
         if self.annotation is AnyStr:
-            return issubclass(str, typ) or issubclass(bytes, typ)
-        return self.annotation is not Any and not self.is_type_var and issubclass(self.annotation, typ)
+            return TypeView(Union[str, bytes]).is_subtype_of(typ)
+        return self.annotation is not Any and not self.is_type_var and self.is_subclass_of(typ)
```

Comment:
tl;dr i've reframed this method in terms of `is_subclass_of`, because i think it's **required** for this method to be useful.

---

I originally left this [comment](https://github.com/litestar-org/type-lens/pull/33#discussion_r1765112431)

but i think i've decided that this method raising makes it **actually** unusable when you're receiving arbitrary input types from a user (given that this library is only useful in the context of receiving unknown input types inside a library).

`TypeView` is ultimately going to be used to implement higher level concepts (in cappa, in litestar, etc) and if you're going to use this method, you're going to **need** to handle the very real cases where a valid user type will fail, being routed through this method.

As such, you'll end up manually wrapping it with a `try` that infers an exception as `False`, which is exactly the same behavior this method should just be doing anyways.

---
And I think the original argument that it returning False for as-yet-unknown cases is bad and a breaking change if it suddenly becomes True. I believe I disagree.

I think it would be perfectly valid (for example) for this method to return `TypeView(TypeVar(bytes, str)).is_subtype_of(str) == True`. In fact `AnyStr` literally **is** this TypeVar and returns True today.

Fixing this deficiency would simply be a more accurate method that handles more cases. but is a very concrete example of where we're currently returning a value instead of `raise`ing, and the value is logically wrong.

And this class as a whole should strive to provide a wholistic API such that downstream code can make a decision off the result of True and not have to change anything in their code as it becomes more accurate.

For example, maybe the appropriate API is to not just reference `.annotation` and provide a `.concrete/.concretize()` that realizes generics into real types.